js/DropZone/DropZoneComponent.js

```diff
@@ -0,0 +1,564 @@
+            input.value = '';
+        });
+
+        // visually hide input - this should ideally be done in the CSS also to prevent a
+        // flash on load (with some consideration for non javascript users)
+        if (!show) {
+            input.style.display = 'none';
+        }
+    }
+
+    /**
+     * Adds the browse files functionality to our DropZone
+     * @param {Element} browse
+     * @param {Element} input
+     */
+    processBrowseNode (browse, input) {
+        browse.addEventListener('click', event => {
+            event.preventDefault();
+            // trigger a synthetic click on the input node which will activate the
+            // native Browse Files interaction
+            input.click();
+        });
+    }
+
+    /**
+     * Update DropZone info node innerHTML
+     * @param {number} id
+     * @param {String} string
+     */
+    updateInfoState (id, string) {
+        const { info, input, node } = this.instanceManager.getInstance(id);
+        const options = this.optionsManager.getInstanceOptions(id);
+        let browse;
+
+        if (info) {
+            info.innerHTML = string;
+            browse = node.querySelector(`.${options.nodeClasses.browse}`);
+
+            if (browse) {
+                this.instanceManager.updateInstance(id, 'browse', browse);
+                this.processBrowseNode(browse, input);
+            }
+        }
+    }
+
+    /**
+     * Update DropZone Html
+     * - remove validation node if we have one
+     * - add / remove files Html
+     * - remove wrapper if we have no files
+     * @param {number} id
+     */
+    updateDropZoneFiles (id) {
+        const { node } = this.instanceManager.getInstance(id);
+        const options = this.optionsManager.getInstanceOptions(id);
+        const files = this.instanceManager.getFiles(id);
+        let wrapper = node.querySelector(`.${options.nodeClasses.wrapper}`);
+        let fileNodeString = '';
+
+        // if we've got a drop we know we don't have any errors
+        // clear any previous validation messages
+        this.validationManager.clear(node, options.nodeClasses.validation);
+
+        // if there are no files we'll remove the wrapper
+        if (!files.length) {
+            if (wrapper) {
+                wrapper.parentNode.removeChild(wrapper);
+            }
+
+            return;
+        }
+
+        // if we do not already have a wrapper, create one
+        if (!wrapper) {
+            wrapper = document.createElement('div');
+            wrapper.className = options.nodeClasses.wrapper;
+            node.appendChild(wrapper);
+        }
+
+        // create file html string
+        files.forEach(file => {
+            fileNodeString += this.utils.createFileNode(file, options);
+        });
+
+        // update wrapper html
+        wrapper.innerHTML = fileNodeString;
+        [].slice.call(wrapper.querySelectorAll(`.${options.nodeClasses.close}`)).forEach(closeButton => {
```

Comment:
```
[].forEach.call(document.querySelectorAll('div'))
VM111476:1 Uncaught TypeError: undefined is not a function
    at NodeList.forEach (<anonymous>)
    at <anonymous>:1:12
```

Doesn't seem to work, I would guess this is because `ForEach` doesn't return anything?